### PR TITLE
e2e: improve log collection

### PR DESCRIPTION
**Pull request by miampf**
Improve the log collection to also collect init container logs. This is done by adding a `log-collector` daemon set to every e2e cluster that saves logs from the host node which then are downloaded to the local machine through the updated `get-logs` script.

### CODE REVIEW

File: packages/scripts.nix [hunk 1]
```diff
@@ -425,30 +425,32 @@
...
+      pod="$(kubectl get pods -o name -n "$namespace" | grep log-collector | cut -c 5-)"
+      mkdir -p ./workspace/logs
+      kubectl wait --for=condition=Ready -n "$namespace" "pod/$pod" 1>/dev/null 2>/dev/null
+      # Download and extract the logs every 3 seconds
+      while true; do
+        kubectl exec -n "$namespace" "$pod" -- /bin/bash -c "rm -f /exported-logs.tar.gz; tar zcvf /exported-logs.tar.gz /export" 1>/dev/null 2>/dev/null
+        rm -f ./workspace/logs/exported-logs.tar.gz
+        kubectl cp -n "$namespace" "$pod:/exported-logs.tar.gz" ./workspace/logs/exported-logs.tar.gz 1>/dev/null 2>/dev/null
+        tar xzvf ./workspace/logs/exported-logs.tar.gz --directory ./workspace/logs 1>/dev/null 2>/dev/null
+        sleep 3
```
Comment (burgerdev) [inline packages/scripts.nix:452]: Why do we need to create tarballs continuously? I imagined that this would happen only after the test finished.

**Review by burgerdev — COMMENTED** [review]
This looks great - I did not think the implementation could be this concise!

File: packages/scripts.nix (hunk 1 above)
Comment (miampf) [reply packages/scripts.nix:452]: I did not find a _good_ way to know if the test is finished or not, except maybe making an explicit download script (or giving `get-logs` CLI parameters like `start` and `download`). Also, I believe that this approach is more stable if the namespace with the log collector accidentally gets cleaned up before logs could be downloaded. If you think that this is unnecessary and has to much overhang, I'll gladly go with the "two scripts/two-part script" solution :)

Comment (burgerdev) [reply packages/scripts.nix:452]: You could handle the TERM signal (or send and intercept a different signal) with logs export.

Comment (miampf) [reply packages/scripts.nix:454]: I'm not sure if I understand what you mean correctly? I can't imagine a way to (easily) let the remote log collector send a signal to a local process and I think it should not be needed to add extra lines of code to every e2e tests go source just to tell the log collector that it is finished. I _could_ try to queue the current process list, grep the first entry that matches a `*.test` pattern and wait for its PID to stop but that seems a bit overcomplicated?

Comment (miampf) [reply packages/scripts.nix]: I just split the script into two parts now, it seemed like the simplest solution to me :)

**Comment by miampf** [comment]
Run with new script structure and working log collection: https://github.com/edgelesssys/contrast/actions/runs/11912439438?pr=985

File: packages/scripts.nix [hunk 2]
```diff
@@ -424,33 +424,53 @@
...
-          if ! [[ -f "$logfile" ]]; then
-            {
-              touch "$logfile" # prevents creation of to much processes
-              # wait for all containers of the pod to come online, then collect the logs
-              kubectl wait pod --all --for=condition=Ready --timeout="-1s" -n "$namespace" "$pod" 1>/dev/null 2>/dev/null
-              kubectl logs -f --all-containers=true -n "$namespace" "$pod" > "$logfile"
-            } &
-          fi
+
+      if [[ -z "''${1-}" ]]; then
```
Comment (burgerdev) [inline packages/scripts.nix:436]:
```suggestion
      if [[ $# < 2 ]]; then
```
... and remove the `-z $2` checks below.

**Review by burgerdev — APPROVED** [review]

**Pull request merged**